File: kotlin-automation/src/test/kotlin/com/myecommerce/automation/driver/api/ApiModels.kt
package com.myecommerce.automation.driver.api

import com.fasterxml.jackson.annotation.JsonProperty

data class ApiProduct(
    val id: Int,
    val title: String,
    val description: String,
    val price: Double,
    @JsonProperty("image_url") val imageUrl: String?,
)

data class ApiProductDetail(
    val id: Int,
    val title: String,
    val description: String,
    val price: Double,
    @JsonProperty("image_url") val imageUrl: String?,
    @JsonProperty("delivery_options") val deliveryOptions: List<ApiDeliveryOption>,
)

data class ApiDeliveryOption(
    val id: Int,
    val name: String,
    val description: String,
    val price: Double,
    @JsonProperty("is_active") val isActive: Boolean,
    @JsonProperty("min_order_amount") val minOrderAmount: Double?,
)

File: typescript-automation/src/driver/api/apiModels.ts
export type ApiProduct = {
  readonly id: number;
  readonly title: string;
  readonly description: string;
  readonly price: number;
  readonly image_url: string | null;
};

export type ApiDeliveryOption = {
  readonly id: number;
  readonly name: string;
  readonly description: string;
  readonly price: number;
  readonly is_active: boolean;
  readonly min_order_amount: number | null;
};

export type ApiProductDetail = {
  readonly id: number;
  readonly title: string;
  readonly description: string;
  readonly price: number;
  readonly image_url: string | null;
  readonly delivery_options: ReadonlyArray<ApiDeliveryOption>;
};
